crates/shared/src/price_estimation/factory.rs

```diff
@@ -247,6 +250,38 @@ impl<'a> PriceEstimatorFactory<'a> {
         }
     }
 
+    fn create_native_estimator(
```

Comment:
I think the idea makes sense but it also illustrates @nlordell's [point](https://github.com/cowprotocol/services/pull/1876#issuecomment-1728968554) about moving all the spot price API into the driver.
That way the core services would eventually only have 1 piece of logic to turn driver URLs into regular price estimators and one for turning them into native price estimators.